src/interceptors/ClientRequest/utils/normalizeHttpRequestParams.test.ts

```diff
@@ -116,6 +116,16 @@ test('handles [RequestOptions, callback] input', () => {
   expect(callback).toHaveProperty('name', 'cb')
 })
 
+test('handles [Empty RequestOptions, callback] input', () => {
+  const [_, __, callback] = normalizeHttpRequestParams(
```

Comment:
Are you sure you can dispatch `new ClientRequest` without any options? That doesn't seem to be a realistic scenario. When would such thing happen in practice?

Reply:
Yes, since each of the properties on `RequestOptions` is optional.

In practice, it's difficult to say whether you would see it used much without _any_ of the options being set, but the point is that since none of them are mandatory, there isn't a single one that can be relied upon, so a sane output must be produced in the case where each option is missing. The easiest way to test this is but setting no options.

I did not test the output, since the URL construction is delegated to another function, so here the test will pass if we do not get an exception. The test for `getUrlByRequestOptions` asserts the correct default URL is produced.